src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/clusterclient/clusterinteraction/ClusterInteractionHistoryFragment.kt

```diff
@@ -0,0 +1,69 @@
+package com.google.chip.chiptool.clusterclient.clusterinteraction
+
+import android.os.Bundle
+import android.util.Log
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.google.chip.chiptool.R
+import kotlinx.android.synthetic.main.cluster_interaction_history_fragment.view.historyCommandList
+import com.google.chip.chiptool.clusterclient.clusterinteraction.ClusterDetailFragment.Companion.endpointId
+
+/**
+ * A simple [Fragment] subclass for the cluster interaction history component
+ * Use the [ClusterInteractionHistoryFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class ClusterInteractionHistoryFragment : Fragment() {
+
+  override fun onCreateView(
+    inflater: LayoutInflater, container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View? {
+    // Inflate the layout for this fragment
+    Log.d(TAG, clusterInteractionHistoryList.toString())
+    return inflater.inflate(R.layout.cluster_interaction_history_fragment, container, false).apply {
+      historyCommandList.adapter =
+        HistoryCommandAdapter(clusterInteractionHistoryList, HistoryCommandListener(), inflater)
+      historyCommandList.layoutManager = LinearLayoutManager(requireContext())
+    }
+  }
+
+  private fun showFragment(fragment: Fragment, showOnBack: Boolean = true) {
+    val fragmentTransaction = requireActivity().supportFragmentManager
+      .beginTransaction()
+      .replace(R.id.fragment_container, fragment, fragment.javaClass.simpleName)
+
+    if (showOnBack) {
+      fragmentTransaction.addToBackStack(null)
+    }
+
+    fragmentTransaction.commit()
+  }
+
+  companion object {
+    /**
+     * Use this factory method to create a new instance of
+     * this fragment using the provided parameters.
+     * @return A new instance of fragment ClusterInteractionHistoryFragment.
+     */
+
+    private const val TAG = "ClusterInteractionHistoryFragment"
+    var clusterInteractionHistoryList = ArrayDeque<HistoryCommand>()
```

Comment:
Should make it very clear that this list is most-recent-first, otherwise a first-time reader has to scan the code to see how this list is used.